pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -827,6 +827,9 @@ func (c *Controller) InitNetworkLookup(meshNetworks *meshconfig.MeshNetworks) {
 
 // return the mesh network for the endpoint IP. Empty string if not found.
 func (c *Controller) endpointNetwork(endpointIP string) string {
+	if len(c.network) != 0 {
+		return c.network
```

Comment:
Donot understand this logic very well. Does it mean all endpointIP belongs to the same network? What if the endpoint is from another network?

Reply:
It means that all endpoints coming from a specific registry belong to configure network (if provided in the networks config with `fromRegistry` value. Otherwise figure from CIDR. I will document it.

Reply:
sounds reasonable